packages/quantic/force-app/main/default/lwc/quanticPager/e2e/quantic-pager.e2e.ts

```diff
@@ -0,0 +1,218 @@
+import {Page} from '@playwright/test';
+import {testSearch, testInsight, expect} from './fixture';
+import {SearchObject} from '../../../../../../playwright/page-object/search-object';
+import {
+  useCaseTestCases,
+  useCaseEnum,
+} from '../../../../../../playwright/utils/use-case';
+import {InsightObject} from '../../../../../../playwright/page-object/insight-object';
+import {ConfigurationObject} from '../../../../../../playwright/page-object/configuration-object';
+
+interface PageObjects {
+  configuration: ConfigurationObject;
+  search: SearchObject;
+  insight?: InsightObject;
+}
+
+interface PagerOptions {
+  useCase: string;
+  numberOfPages: number;
+}
+
+async function visitPage(
+  page: Page,
+  {search, insight, configuration}: PageObjects,
+  options: Partial<PagerOptions> = {}
+) {
+  const pageUrl = 's/quantic-pager';
+  await page.goto(pageUrl);
+  configuration.configure(options);
+  if (options.useCase === useCaseEnum.insight) {
+    await insight?.waitForInsightInterfaceInitialization();
+    await search.interceptSearchAndLimitResultPages(5);
+    await search.performSearch();
+    await search.waitForSearchResponse();
+  } else {
+    await search.interceptSearchAndLimitResultPages(5);
+    await search.waitForSearchResponse();
+  }
+}
+
+async function visitPageAndloadOptionsFromUrl(
+  page: Page,
+  {search, configuration}: PageObjects,
+  urlHash: string
+) {
+  const pageUrl = 's/quantic-pager';
+  await page.goto(`${pageUrl}#${urlHash}`);
+  await search.interceptSearchAndLimitResultPages(5);
+  configuration.configure();
+}
+
+async function setupWithPauseBeforeSearch(
+  page: Page,
+  {search, insight, configuration}: PageObjects,
+  options: Partial<PagerOptions> = {}
+) {
+  const pageUrl = 's/quantic-pager';
+  await page.goto(pageUrl);
+
+  if (options.useCase === useCaseEnum.insight) {
+    configuration.configure(options);
+    await insight?.waitForInsightInterfaceInitialization();
+    const stalledRequest = search.interceptSearchIndefinitely();
+    await search.performSearch();
+    return stalledRequest;
+  }
+  const stalledRequest = search.interceptSearchIndefinitely();
+  configuration.configure(options);
+  return stalledRequest;
+}
```

Comment:
what do you mean exactly?  a specific  file `functions.ts`  under `quanticPager/e2e/*` ?

Reply:
I mean like a `functions` folder under your playwright general folder for helper functions that could be re-used through all the tests.

Reply:
The code has change a bit since your last comment, so let me know when you think now, 
Basically now the setup steps are fully encapsulated in the fixtures.